pkg/tcpip/transport/tcp/cubic.go

```diff
@@ -66,6 +118,56 @@ func (c *cubicState) enterCongestionAvoidance() {
 	}
 }
 
+// updateHyStart tracks packet round-trip time (rtt) to find a safe threshold
+// to exit slow start without triggering packet loss.  It updates the SSThresh
+// and sets FoundThresh when it does.
+func (c *cubicState) updateHyStart(rtt time.Duration) {
+	if rtt < 0 {
+		// negative indicates unknown
+		return
+	}
+	now := c.s.ep.stack.Clock().NowMonotonic()
+	if c.EndSeq.LessThan(c.s.SndUna) {
+		c.beginHyStartRound(now)
+	}
+	// ACK train
+	if now.Sub(c.LastAck) < ackDelta && // ensures acks are part of the same "train"
+		c.LastRTT < effectivelyInfinity {
```

Comment:
Optional/style: the `if` condition can be one line if you prefer.

Reply:
I prefer this style since there is a comment on the first clause, but will defer to project code style that applies here.